### PR TITLE
change to more idiomatic python in tutorial 101

### PR DESCRIPTION
**Pull request by michhar**
Changes:
```python
from cntk import *
```
will be replaced with 
```python
from cntk import input_variable, parameter, times, \
                 cross_entropy_with_softmax, classification_error, \
                 learning_rate_schedule, sgd, Trainer, UnitType, \
                 softmax
```

Fixed IndexError found in:
```python
print("Predicted:", [np.argmax(result[i,:]) for i in range(len(result))])
```
and replaced with
```python
print("Predicted:", [np.argmax(x) for x in result[0]])
```

----------
The dictionary declaration:
```python
# Define a dictionary to store the model parameters
mydict = {"w":None,"b":None} 
```
Can be better written:
```
# Define a dictionary to store the model parameters
mydict = {}
```

Under Run the trainer using defaultdict for ease of use for appending to a dict of arrays
```python
# Run the trainer and perform model training
training_progress_output_freq = 50

plotdata = {"batchsize":[], "loss":[], "error":[]}
```

can be better written:
```python
from collections import defaultdict

# Run the trainer and perform model training
training_progress_output_freq = 50
plotdata = defaultdict(list)
```

Jupyter notebook tested with both python 2.7 and 3.5 on Windows 10 with cntk '2.0rc2' CPU-only.

### CODE REVIEW

**Comment by sayanpa** [comment]
Please update the tutorials addressing the comments.

File: Tutorials/CNTK_101_LogisticRegression.ipynb [hunk 1]
```diff
@@ -145,15 +145,20 @@
...
+   },
    "outputs": [],
    "source": [
     "# Import the relevant components\n",
     "from __future__ import print_function\n",
     "import numpy as np\n",
     "import sys\n",
     "import os\n",
-    "from cntk import *\n",
+    "from cntk import input_variable, parameter, times, \\\n",
```
Comment (michhar) [inline Tutorials/CNTK_101_LogisticRegression.ipynb]: Done.  Thanks for the heads up.

File: Tutorials/CNTK_101_LogisticRegression.ipynb [hunk 2]
```diff
@@ -145,15 +145,20 @@
   {
    "cell_type": "code",
    "execution_count": 5,
```
Comment (michhar) [inline Tutorials/CNTK_101_LogisticRegression.ipynb]: Done.  Please let me know if there are still any problems.  Jupyter notebooks can be tricky here.

**Comment by michhar** [comment]
@sayanpa I have addressed these.  Let me know if you approve/not.  Ideally I'd like to run a jupyter notebook version of the pep8 check, flake8.  There are a few items we can clean up more, but will save for future PR. :)  Thanks for your comments and considering these changes.

**Comment by sayanpa** [comment]
Would you mind squashing all the commits into a single pull request, and I will integrate. Alternatively, I can do it for you but your check-in history will show up under my name. I would rather you have the credit for the fix.

**Comment by michhar** [comment]
@sayanpa I will do this and close this PR.

**Pull request closed without merging**